src/Compilers/CSharp/Portable/Symbols/TypeSymbolExtensions.cs

```diff
@@ -1964,5 +1965,13 @@ internal static int TypeToIndex(this TypeSymbol type)
                 default: return -1;
             }
         }
+
+        internal static void CheckRecordLangVersion(this TypeSymbol type, DiagnosticBag diagnostics, TypeSyntax syntax, Location location)
+        {
+            if (type.IsErrorType() && syntax is IdentifierNameSyntax name && name.Identifier.Text == SyntaxFacts.GetText(SyntaxKind.RecordKeyword))
```

Comment:
>type.IsErrorType() [](start = 16, length = 18)

Would it be possible to be more precise in detecting the situation when the type is not found?

Reply:
Perhaps `ErrorTypeSymbol.ResultKind`?